src/components/Gallery/components/GalleryHeader/GalleryHeader.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+
+import {ArrowLeft, ArrowRight, Xmark} from '@gravity-ui/icons';
+import type {ButtonProps} from '@gravity-ui/uikit';
+import {ActionTooltip, Button, Flex, Icon, Text, useDirection, useMobile} from '@gravity-ui/uikit';
+
+import {block} from '../../../utils/cn';
+import type {GalleryItemAction} from '../../GalleryItem';
+import {i18n} from '../../i18n';
+import {MobileGalleryHeader} from '../MobileGalleryHeader/MobileGalleryHeader';
+import {FullScreenAction} from '../actions';
+
+import './GalleryHeader.scss';
+
+const cnGalleryHeader = block('gallery-header');
+
+export type GalleryHeaderProps = {
+    itemName?: React.ReactNode;
+    actions?: GalleryItemAction[];
+    withNavigation: boolean;
+    activeItemIndex: number;
+    itemsLength: number;
+    fullScreen: boolean;
+    onBackClick: () => void;
+    onGoToPrevious: () => void;
+    onGoToNext: () => void;
+    onUpdateFullScreen: React.Dispatch<React.SetStateAction<boolean>>;
+    onClose: () => void;
+    hidden?: boolean;
+    interactive?: boolean;
+};
+
+export const GalleryHeader = ({
+    itemName,
+    actions,
+    withNavigation,
+    activeItemIndex,
+    itemsLength,
+    fullScreen,
+    onBackClick,
+    onGoToPrevious,
+    onGoToNext,
+    onUpdateFullScreen,
+    onClose,
+    hidden,
+    interactive,
+}: GalleryHeaderProps) => {
+    const direction = useDirection();
+    const isMobile = useMobile();
+
+    if (isMobile) {
+        return (
+            <MobileGalleryHeader
+                itemName={itemName}
+                actions={actions}
+                withNavigation={withNavigation}
+                activeItemIndex={activeItemIndex}
+                itemsLength={itemsLength}
+                onBackClick={onBackClick}
+                hidden={hidden}
+                interactive={interactive}
+            />
+        );
+    }
+
+    return (
+        <Flex
```

Comment:
Maybe is would be better to create a component for the desktop version too and use `GalleryHeader` for choosing the view